worker/src/lib/providerCalls/call.ts

```diff
@@ -38,9 +39,11 @@ export async function callProvider(props: CallProps): Promise<Response> {
     props;
   const apiBaseUrl = new URL(apiBase.replace(/\/$/, "")); // remove trailing slash if any
 
-  const new_url = new URL(
-    `${apiBaseUrl.origin}${originalUrl.pathname}${originalUrl.search}`
-  );
+  const new_url = azurePattern.test(apiBase)
+    ? apiBaseUrl
```

Comment:
We dont want this behavior because a lot of libraries will auto tack on the /chat or /completion endpoints to the baseURL.. 

We need to instead change the logic to append it like this

`${apiBaseUrl.href}${originalUrl.pathname}${originalUrl.search}`

but this is might break if there are two `/v1` paths